File: security_assistant/remediation/code_examples/xss_fix.go
// VULNERABLE:
// fmt.Fprintf(w, "<div>%s</div>", userInput)

// SECURE (HTML Template with Auto-Escaping):
import "html/template"

tmpl := template.Must(template.New("page").Parse("<div>{{.}}</div>"))
tmpl.Execute(w, userInput) // Automatically escapes HTML

File: security_assistant/remediation/code_examples/xss_fix.py
# VULNERABLE:
# return "<h1>Hello, " + user_input + "</h1>"

# SECURE (using html.escape):
import html

safe_input = html.escape(user_input)
return f"<h1>Hello, {safe_input}</h1>"

# SECURE (using Flask/Jinja2 - automatic):
# {{ user_input }}  <-- Jinja2 escapes by default
